pkg/tfbridge/schema.go

```diff
@@ -351,6 +351,12 @@ func (ctx *conversionContext) makeTerraformInput(
 	tfs shim.Schema,
 	ps *SchemaInfo,
 ) (interface{}, error) {
+
+	if v.IsSecret() {
+		// We shouldn't see secrets, but if we do then we should ignore them
```

Comment:
This is somewhat reasonable, because we can't transform first-class secrets into TF we should not be handling them. There is though the need to handle secrets through black-box TF transformations, and it's been done elsewhere .. I'm wondering how you're triggering this code path where secrets are actually pushed in here, and should't the outer secret handling take care of that, if it's not taking care of the secrets perhaps it's also not secreting the results properly on the output side.

Reply:
This is where I am testing it https://github.com/pulumi/pulumi-terraform-bridge/blob/b2e46fef4662978fc9b924fafe7f930bc837b0c4/pkg/tfbridge/provider_test.go#L5738-L5753

Also see https://pulumi.slack.com/archives/C02FXTZEZ6W/p1723217439575279?thread_ts=1723201028.986739&cid=C02FXTZEZ6W

Reply:
I've added a suggestion to strip secrets specifically in the newly added code, it's more obviously safe there.